pkg/kube/tunnel.go

```diff
@@ -62,7 +62,7 @@ func (t *Tunnel) Close() {
 }
 
 // ForwardPort opens a tunnel to a kubernetes pod
-func (t *Tunnel) ForwardPort() error {
+func (t *Tunnel) ForwardPort(localPort int) error {
```

Comment:
any way we can add `localPort` as a field to the `Tunnel` struct instead? That way we aren't breaking function signatures in pkg/, which we sorta consider as backwards-incompatible changes and may make it harder to make the argument to merge this in before Helm 3.

Reply:
`Tunnel` has a filed named `Local`, so I can set it after `kube.NewTunnel`. However, I have to pass the `localPort` to `New` function in `portforwarder.go`, is that ok? 

```go
// New creates a new and initialized tunnel.
func New(namespace string, client kubernetes.Interface, config *rest.Config, localPort int) (*kube.Tunnel, error) {   // pass localPort here
	podName, err := getTillerPodName(client.CoreV1(), namespace)
	if err != nil {
		return nil, err
	}
	const tillerPort = 44134
	t := kube.NewTunnel(client.Core().RESTClient(), config, namespace, podName, tillerPort)
	t.Local = localPort   //set local port to tunnel here.
	return t, t.ForwardPort()
}
```

Reply:
I assume passing to the `New` functions still qualifies as a breaking change.

How about adding a new method in the package, `ForwardToLocalPort(localPort int) error {}` and keeping the existing `ForwardPort` method?

Reply:
By setting `t.Local` to `localPort` could avoid adding the new method `ForwardToLocalPort`, because the `localPort` param could be passed to `ForwardPort` with `t`(see the codes above).

But I still have to pass the local port from `setupConnection` in cmd/helm/helm.go to `portforwarder.New` in `pkg/helm/portforwarder/portforwarder.go`. 

How about add a new method `NewWithLocalPort`? 

```go
// NewWithLocalPort creates a new and initialized tunnel.
func NewWithLocalPort(namespace string, client kubernetes.Interface,
 config *rest.Config, localPort int) (*kube.Tunnel, error) {
	podName, err := getTillerPodName(client.CoreV1(), namespace)
	if err != nil {
		return nil, err
	}
	const tillerPort = 44134
	t := kube.NewTunnel(client.Core().RESTClient(), config, namespace, podName, tillerPort)
	t.Local = localPort   //set local port to tunnel here.
	return t, t.ForwardPort()
}
```